reckless_github/src/repository.rs

```diff
@@ -57,68 +51,56 @@ impl Github {
-                                PluginLang::TypeScript,
-                            ));
-                        }
-                        "reckless.yml" | "reckless.yaml" => {
-                            let conf_path = format!("{}/{}", path_to_plugin, file_name);
-                            let mut conf_str = String::new();
-                            File::open(conf_path.as_str())
-                                .await?
-                                .read_to_string(&mut conf_str)
-                                .await?;
-                            let _: Conf = serde_yaml::from_str(&conf_str).unwrap();
-                            // FIXME: store the conf inside the plugin
-                        }
-                        _ => continue,
+        // We are filtering to not iterate through files or any directories such as .git, .ci, .github
+        for plugin_dir in WalkDir::new(repo_path)
+            .max_depth(1)
+            .into_iter()
+            .filter_entry(|dir_entry| {
+                dir_entry
+                    .file_name()
+                    .to_str()
+                    .map(|s| {
+                        dir_entry.depth() == 0 || !s.starts_with(".") && dir_entry.path().is_dir()
+                    })
+                    .unwrap()
+            })
+        {
+            match plugin_dir {
+                Ok(plugin_path) => {
+                    let mut conf = ();
+                    for file in WalkDir::new(plugin_path.path().to_str().unwrap()).max_depth(1) {
+                        let file_dir = file.unwrap().clone();
+                        let (path_to_plugin, plugin_name) =
+                            get_plugin_info_from_path(file_dir.path()).unwrap();
+                        let plugin_lang = match file_dir.file_name().to_str().unwrap() {
+                            "requirements.txt" => PluginLang::Python,
+                            "go.mod" => PluginLang::Go,
+                            "cargo.toml" => PluginLang::Rust,
+                            "pubspec.yaml" => PluginLang::Dart,
+                            "package.json" => PluginLang::JavaScript,
+                            "tsconfig.json" => PluginLang::TypeScript,
+                            /*
+                            FIXME: WHAT NEEDS TO BE DONE HERE??
+                            "reckless.yml" | "reckless.yaml" => {
+                                let conf_path = format!("{}/{}", path_to_plugin, file_name);
+                                let mut conf_str = String::new();
+                                File::open(conf_path.as_str())
+                                    .await?
+                                    .read_to_string(&mut conf_str)
+                                    .await?;
+                                let _: Conf = serde_yaml::from_str(&conf_str).unwrap();
+                                // FIXME: store the conf inside the plugin
+                            }
+                             */
```

Comment:
I am not sure what needs to be done with this code?

Reply:
This should load the plugin manifest that we encode into yaml format